dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/interpretation/impl/DefaultInterpretationService.java

```diff
@@ -157,6 +167,27 @@ public void deleteInterpretation( Interpretation interpretation )
         return interpretationStore.getAllOrderedLastUpdated( first, max );
     }
 
+    @Override
+    public List<Mention> updateMentions( String text )
+    {
+        List<Mention> mentions = new ArrayList<Mention>();
+
+        Matcher matcher = Pattern.compile( "^[@]\\w+|(?<=\\s)[@]\\w+" ).matcher( text );
```

Comment:
I am not familiar with the lookbehind operator, why do we need it here? We could use capturing parenthesis. My first attempt would be `"(?:\\s|^)@([\\w]+)"` and then `String group = matcher.group(1);`.

In any case, there is a catch here, a username can contain any char (I just created one named `0@test1!""·$·$%&/()?¿_:;{}*/-+S<>&'"`...). And now consider punctuation or exclamation sign used after a username, i.e. `@user,  please inform @otheruser!`. Do we limit usernames to alpha and `-_.` or something like that? `"(?:\\s|^)@([\\w+._-]+)"`

Reply:
You are completely right. I have changed the code to use your regexp. It is still a bit strange to not allow any sort of username but it is too tricky to parse any character. Anyway, I agree that at least '-_.' should be allowed for mentioning. They are used everywhere. We can refine it in the next iteration